lib/controller.js

```diff
@@ -705,39 +710,152 @@ controller.disableAccessPoint = function(opts) {
   });
 };
 
+// clear terminal windows and linux
+controller.clear = function() {
+  return new Promise(function(resolve) {
+    // clears terminal
+    if (!global.IS_TEST_ENV) {
+      process.stdout.write('\u001b[2J\u001b[0;0H');
+    }
+    resolve();
+  });
+};
+
+// ssh connection with advanced options and error handling
+controller.remote = function(tessel, opts) {
+  logs.info('Starting remote SSH Session on Tessel at ' + opts.ip + '.\nType \'exit\' at the prompt to end.');
+  return new Promise(function(resolve, reject) {
+    var options = ['-i',
+      opts.key,
+      '-p',
+      opts.port,
+      '-e',
+      opts.escape_char,
+      opts.username + '@' + tessel.lanConnection.ip
+    ];
+
+    var ch = require('child_process')
+      .spawn('ssh', options, {
+        stdio: 'inherit'
+      });
+
+    // error handling
+    ch.once('error', function(e) {
+      if (e === 255) {
+        controller.clear()
+          .then(function() {
+            logs.warn('Login to Tessel at failed with KEY ' + opts.key + ' !');
+          }).then(resolve);
+      } else {
+        reject('Error while connected to ' + tessel.name + ':', e);
+      }
+    });
+
+    // exit handling
+    ch.once('exit', function(e) {
+      if (e === 127 || e === 0) { // exit by user
+        // clear console
+        controller.clear()
+          .then(function() {
+            logs.warn('Connection to ' + tessel.name + ' closed!\n');
+            resolve();
+          });
+      } else if (e === 255) {
+        controller.clear()
+          .then(function() {
+            logs.warn('You are not authorized with key ' + opts.key);
+          }).then(resolve);
+      } else {
+        controller.clear()
+          .then(function() {
+            logs.warn('Connection to ' + tessel.name + ' closed due to reason (error code): ', e);
```

Comment:
I recommend changing the end of this string to `"closed:", e).` because `due to reason` doesn't make perfect sense.